### DISCUSSION TITLE
resource not available for integration

### DISCUSSION THREAD

**Discussion by AkashKumar7902** [rossjrw/pr-preview-action#59] [Q&A] [answered]
getting errors on : https://github.com/officialasishkumar/docs/actions/runs/6205862185/job/16849480400

**Comment by rossjrw** [comment]
(moved this from an Issue to a Discussion, as it's a request for help and not a bug report/feature request)

From the linked log:

```
remote: Permission to officialasishkumar/docs.git denied to github-actions[bot].
fatal: unable to access 'https://github.com/officialasishkumar/docs.git/': The requested URL returned error: 403
```

Looks like the action runner can't access the repository. Could be an issue with your GITHUB_TOKEN.

The GITHUB_TOKEN is set automatically, but I see that you've hardcoded one from your repo secrets with `env: GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}`. Is that definitely the right token to use? Try using the default one - your repo is public so you shouldn't need any workarounds, as far as I'm aware.

https://github.com/officialasishkumar/docs/blob/a8f96c120eeb1a8b5129951178cdcf22b3bbda17/.github/workflows/build_and_check.yml#L24-L26

**Reply by AkashKumar7902** [reply]
@rossjrw Still the same issue 
https://github.com/officialasishkumar/docs/actions/runs/6212903398/job/16863492671?pr=9

Edit: the same error is in the main repo of the fork https://github.com/keploy/docs/actions/runs/6207657330/job/16853059594

**Reply by rossjrw** [reply]
@AkashKumar7902 Looking at your pull request (https://github.com/officialasishkumar/docs/pull/9), the error is occurring because the request originates from a fork (https://github.com/AkashKumar7902/docs). This Action does not currently support previews from forks. This is mentioned in the README.

See here for explanation of why, and some workarounds in case you need this feature despite security vulnerabilities: https://github.com/rossjrw/pr-preview-action/pull/6

**Reply by AkashKumar7902** [reply]
@rossjrw 
I have linked an action take occured in the main repo with the same workflow file but still it throws an error

**Reply by rossjrw** [reply]
@AkashKumar7902 can you link the corresponding pull request?

**Comment by AkashKumar7902** [comment]
@rossjrw https://github.com/keploy/docs/actions/runs/6207657330/job/16853059594

**Reply by rossjrw** [reply]
@AkashKumar7902 This is an actions workflow log. Can you please link the pull request?

**Reply by AkashKumar7902** [reply]
@rossjrw 
https://github.com/keploy/docs/pull/198

**Reply by rossjrw** [reply]
@AkashKumar7902 This is also a pull request made from a fork. The source repository is https://github.com/keploy/docs and the fork is https://github.com/AkashKumar7902/docs.

![image](https://github.com/rossjrw/pr-preview-action/assets/29130152/8e732572-67ac-481e-8150-6cb812435862)

An example of a pull request *not* made from a fork is https://github.com/keploy/docs/pull/193, which has both branches in the same repository.

**Reply by AkashKumar7902** [reply]
@rossjrw after the merge the following PRs were also unable to deploy to github pages because of the same error.
Example : https://github.com/keploy/docs/pull/203

**Reply by rossjrw** [reply, marked as answer]
@AkashKumar7902 The linked pull request also originates from a fork: https://github.com/ganeshrevadi/docs

![image](https://github.com/rossjrw/pr-preview-action/assets/29130152/c9185d99-b8de-488b-92bb-b1bac5603239)

Please review to confirm your understanding of what a fork is: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/about-forks

Just to be clear, it looks like this is something for the Keploy docs maintainers to address. It looks like their workflows have been failing for several weeks, though, so it doesn't look like this is a priority for them.
